File: server/src/main/java/de/davelee/trams/server/request/MessageRequest.java
package de.davelee.trams.server.request;

/**
 * This class is part of the TraMS Server REST API. It represents a request to add a message.
 * @author Dave Lee
 */
public class MessageRequest {

    /**
     * The company receiving this message.
     */
    private String company;

    /**
     * The subject of the message.
     */
    private String subject;

    /**
     * The content of the message.
     */
    private String text;

    /**
     * The sender of this message.
     */
    private String sender;

    /**
     * The folder where this message is located.
     */
    private String folder;

    /**
     * The date and time that this message was sent in format dd-MM-yyyy HH:mm.
     */
    private String dateTime;

    public MessageRequest() {
    }

    public MessageRequest(String company, String subject, String text, String sender, String folder, String dateTime) {
        this.company = company;
        this.subject = subject;
        this.text = text;
        this.sender = sender;
        this.folder = folder;
        this.dateTime = dateTime;
    }

    public String getCompany() {
        return company;
    }

    public void setCompany(String company) {
        this.company = company;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getSender() {
        return sender;
    }

    public void setSender(String sender) {
        this.sender = sender;
    }

    public String getFolder() {
        return folder;
    }

    public void setFolder(String folder) {
        this.folder = folder;
    }

    public String getDateTime() {
        return dateTime;
    }

    public void setDateTime(String dateTime) {
        this.dateTime = dateTime;
    }

    @Override
    public String toString() {
        return "MessageRequest{" +
                "company='" + company + '\'' +
                ", subject='" + subject + '\'' +
                ", text='" + text + '\'' +
                ", sender='" + sender + '\'' +
                ", folder='" + folder + '\'' +
                ", dateTime='" + dateTime + '\'' +
                '}';
    }
}

File: desktop/src/app/messages/message.request.ts
/**
 * This class defines a model for requests to create messages in TraMS.
 */
export class MessageRequest {

    private company: string;
    private subject: string;
    private text: string;
    private sender: string;
    private folder: string;
    private dateTime: string;

    /**
     * Create a new message request.
     * @param company the company that the message belongs to as a string.
     * @param subject the subject of the message as a string.
     * @param content the content of the message as a string.
     * @param sender the sender of the message as a string.
     * @param folder the folder of the message as a string.
     * @param date the date of the message as a string in the format dd-MM-yyyy HH:mm.
     */
    constructor(company: string, subject: string, content: string, sender: string, folder: string, date: string) {
        this.company = company;
        this.subject = subject;
        this.text = content;
        this.sender = sender;
        this.folder = folder;
        this.dateTime = date;
    }

    /**
     * Retrieve the subject of the message.
     * @return the subject as a string.
     */
    getSubject(): string {
        return this.subject;
    }

    /**
     * Retrieve the content of the message.
     * @return the content as a string.
     */
    getContent(): string {
        return this.text;
    }

    /**
     * Retrieve the folder of the message.
     * @return the folder as a string.
     */
    getFolder(): string {
        return this.folder;
    }

    /**
     * Retrieve the date and time of the message.
     * @return the date and time as a string in the format dd-MM-yyyy HH:mm.
     */
    getDateTime(): String {
        return this.dateTime;
    }

    /**
     * Retrieve the sender of the message.
     * @return the sender as a string.
     */
    getSender(): string {
        return this.sender;
    }

}